### PR TITLE
Playlist creating

### PR DESCRIPTION
**Pull request by medegor44**

### CODE REVIEW

File: src/App.jsx [hunk 1]
```diff
@@ -4,36 +4,51 @@ import SpotifyLogin from "./SpotifyLogin";
...
-            <div className="headContainer">
-              <h1>Spotify playlist generator</h1>
-              <SpotifyLogin authorized={authorized} />
+          <UserContext.Provider value={userData}>
+            <div className="container">
+              <div className="headContainer">
+                <h1>Spotify playlist generator</h1>
+                <SpotifyLogin authorized={authorized} />
+              </div>
+              <SpotifySongSearch authorized={authorized} />
```
Comment (confirmit-horizons) [inline src/App.jsx:23]: you can remove authorized prop and replace it with contest.user != null

File: src/App.jsx [hunk 2]
```diff
@@ -4,36 +4,51 @@ import SpotifyLogin from "./SpotifyLogin";
...
-import { getAccessTokenFromLocationHash } from "./utils/spotify";
+import { getAccessTokenFromLocationHash, fetchUser } from "./utils/spotify";
 import "./css/App.css";
+import UserContext from "./contexts/UserContext";
 
 const App = () => {
   const [authorized, setAuthorized] = useState(false);
+  const [userData, setUserData] = useState(null);
 
   useEffect(() => {
```
Comment (confirmit-horizons) [inline src/App.jsx]: good candidate for extraction to separate hook (ie useAuthorization)

File: src/CreatePlaylistButton.jsx [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+        type="text"
+        value={playlistName}
+        onChange={(e) => {
+          setName(e.target.value);
+        }}
+      />
+      <button type="button" onClick={handleButtonClick}>
+        Create playlist
+      </button>
+      {message}
```
Comment (confirmit-horizons) [inline src/CreatePlaylistButton.jsx]: Can be a component instead

**Review by confirmit-horizons — APPROVED** [review]

File: src/App.jsx (hunk 1 above)
Comment (medegor44) [reply src/App.jsx:23]: If I remove authorized prop, then I cannot detect when to show loading indicator. I think, I have to give it more suitable name than autjorized.

File: src/utils/spotify.js [hunk 4]
```diff
@@ -39,9 +41,26 @@ const fetchSpotifyApiData = async (url, token) => {
   return data;
 };
 
+export const createPlaylist = async (token, userId, playlistName) => {
```
Comment (confirmit-horizons) [inline src/utils/spotify.js:44]: Also make sure you create Private playlist by default not Public. (Public playlist visible to anyone, private - visible only for the owner)

File: src/App.jsx (hunk 1 above)
Comment (confirmit-horizons) [reply src/App.jsx:23]: ok, we'll change it later during refactoring

**Review by confirmit-horizons — APPROVED** [review]

**Pull request merged**